app/src/androidTest/java/testapp/NavigationViewOrientationTest.java

```diff
@@ -0,0 +1,53 @@
+package testapp;
+
+import android.content.res.Configuration;
+import android.support.test.espresso.ViewAction;
+
+import com.mapbox.services.android.navigation.testapp.test.TestNavigationActivity;
+
+import org.junit.Test;
+
+import testapp.activity.BaseNavigationActivityTest;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.matcher.ViewMatchers.isRoot;
+import static testapp.action.OrientationChangeAction.orientationLandscape;
+import static testapp.action.OrientationChangeAction.orientationPortrait;
+
+public class NavigationViewOrientationTest extends BaseNavigationActivityTest {
+
+  @Override
+  protected Class getActivityClass() {
+    return TestNavigationActivity.class;
+  }
+
+  @Test
+  public void onOrientationLandscape_navigationContinuesRunning() {
+    if (checkOrientation(Configuration.ORIENTATION_LANDSCAPE)) {
+      return;
+    }
+    validateTestSetup();
+
+    changeOrientation(orientationLandscape());
+  }
+
+  @Test
+  public void onOrientationPortrait_navigationContinuesRunning() {
+    if (checkOrientation(Configuration.ORIENTATION_PORTRAIT)) {
+      return;
+    }
+    validateTestSetup();
+
+    changeOrientation(orientationPortrait());
+  }
+
+  // TODO create test rule for this to conditionally ignore
```

Comment:
Let's revisit `Assumptions and Conditional Test Execution` in a future PR.
For reference 👀 https://www.codeaffine.com/2013/11/18/a-junit-rule-to-conditionally-ignore-tests/